homeassistant/components/smarttub/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""SmartTub integration."""
+import logging
+
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN, SMARTTUB_CONTROLLER
+from .controller import SmartTubController
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass, _config):
+    """Set up smarttub component."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a smarttub config entry."""
+
+    controller = SmartTubController(hass)
+    hass.data[DOMAIN][entry.unique_id] = {
+        SMARTTUB_CONTROLLER: controller,
+    }
+
+    if not await controller.async_setup_entry(entry):
+        raise ConfigEntryNotReady
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass, entry):
+    """Remove a smarttub config entry."""
```

Comment:
Unloading can fail.  We usually do something like this.

```
async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
    """Unload a config entry."""
    unload_ok = all(
        await asyncio.gather(
            *[
                hass.config_entries.async_forward_entry_unload(entry, component)
                for component in PLATFORMS
            ]
        )
    )

    if unload_ok:
        hass.data[DOMAIN].pop(entry.entry_id)

    return unload_ok

```

Reply:
Added with test